src/core/core.scale.js

```diff
@@ -80,7 +80,11 @@ function getLineValue(scale, index, offsetGridLines) {
 	var lineValue = scale.getPixelForTick(index);
 
 	if (offsetGridLines) {
-		if (index === 0) {
+		if (scale.getTicks().length === 1) {
+			lineValue -= scale.isHorizontal() ?
```

Comment:
It'd be an easy change:

```
var xLineValue = getPixelForGridLine(me, index, gridLines.offsetGridLines);
if (helpers.isNullOrUndef(xLineValue)) {
    return;
}
```

Reply:
This is a bit more complicated. If it returns here, not only the grid line but also the tick line and label will disappear as they are drawn together. That's why `lineColor` is set to transparent if the grid line is out of the range.

So, in order to avoid drawing the grid line over axis, we need to check whether each grid line is on the axis (this requires to check `gridLines.drawBorder` and `options.position`) and to set `lineColor` to transparent if needed.

Reply:
Ah, right you are. Ok, in that case, I agree with you that it's complicated enough that it would perhaps be better handled in another PR instead. Thanks for the clarification